arangod/Agency/Supervision.cpp

```diff
@@ -1484,6 +1484,12 @@ void Supervision::checkBrokenCollections() {
     std::shared_ptr<Node> const& db = dbpair.second;
 
     for (auto const& collectionPair : db->children()) {
+      // collectionPair.first is collection id
+      std::pair<std::string, bool> collectionNamePair = collectionPair.second->hasAsString(StaticStrings::DataSourceName);
+      if (!collectionNamePair.second || collectionNamePair.first.front() == '_') {
```

Comment:
From https://en.cppreference.com/w/cpp/string/basic_string/front:
> Returns reference to the first character in the string. The behavior is undefined if empty() == true.

Should we add a check here to make sure that the string is not empty?

Reply:
If `collectionNamePair.first` is really the "collection id" as it is claimed in the comment, then it will not start with an underscore, but rather with a numeric digit.